framework/src/restart/Restartable.C

```diff
@@ -15,13 +15,17 @@
 #include "InputParameters.h"
 #include "Restartable.h"
 #include "SubProblem.h"
+#include "FEProblem.h"
 
 Restartable::Restartable(const InputParameters & parameters, std::string system_name, SubProblem * subproblem) :
     _restartable_name(parameters.get<std::string>("_object_name")),
     _restartable_params(&parameters),
     _restartable_system_name(system_name),
     _restartable_tid(parameters.isParamValid("_tid") ? parameters.get<THREAD_ID>("_tid") : 0),
-    _restartable_subproblem(parameters.isParamValid("_subproblem") ? parameters.get<SubProblem *>("_subproblem") : (parameters.isParamValid("_fe_problem") ? (SubProblem*)parameters.get<FEProblem *>("_fe_problem") : subproblem) )
+    _restartable_subproblem(parameters.isParamValid("_subproblem") ?
+    parameters.get<SubProblem *>("_subproblem") : (parameters.isParamValid("_fe_problem_base") ?
+    parameters.get<FEProblemBase *>("_fe_problem_base") : (parameters.isParamValid("_fe_problem") ?
+    parameters.get<FEProblem *>("_fe_problem") : subproblem)) )
```

Comment:
This needs better indentation to increase legibility. I propose:
```
_var(bool1 ? long_thing :
  (bool2 ? longer_thing :
    (bool3 ? the_longer_things : what ever)))
```
But anything else that will clearly show what is going on with the nesting will work too. Also, since it is a pointer a nested `if` in the `c-tor` could look a lot better.